filters/include/pcl/filters/impl/extract_indices.hpp

```diff
@@ -55,9 +55,9 @@ pcl::ExtractIndices<PointT>::filterDirectly (PointCloudPtr &cloud)
 
   std::vector<pcl::PCLPointField> fields;
   pcl::for_each_type<FieldList> (pcl::detail::FieldAdder<PointT> (fields));
-  for (int rii = 0; rii < static_cast<int> (removed_indices_->size ()); ++rii)  // rii = removed indices iterator
+  for (const auto& rii : (*removed_indices_)) // rii = removed indices iterator
   {
-    std::size_t pt_index = (std::size_t) (*removed_indices_)[rii];
+    std::size_t pt_index = (std::size_t) rii;
```

Comment:
Should be `uindex_t`?

Reply:
Yes, that makes a bit more sense. Unsigned to avoid sign compare warnings. Changed it.